app/views/admin/users/widgets/_roles.html.erb

```diff
@@ -0,0 +1,41 @@
+<header class="mb-4">
+  <h2 class="crayons-subtitle-2 mb-1 flex items-center">
+    <%= crayons_icon_tag(:badge, title: "Assigned roles", class: "mr-2") %>
+    Assigned roles
+  </h2>
+  <p class="color-secondary">Assign a roles to this member based on permissions. <a href="#" class="c-link c-link--branded">Learn more</a></p>
+</header>
+<% unless @user.roles.any? %>
+  <div class="flex justify-between items-center">
+    <p>No special role assigned yet.</p>
+    <button class="c-btn c-btn--secondary whitespace-nowrap" type="button">Add role</button>
+  </div>
+<% else %>
+  <ul class="flex flex-wrap gap-2">
+    <% @user.roles.each do |role| %>
+      <% if role.name == "banned" || role.name == "suspended" || role.name == "super_admin" || @user.id == current_user.id %>
+        <% if role.resource_name %>
+          <%= role.resource_name.to_s %>
+        <% else %>
+          <%= role_display_name(role) %>
+        <% end %>
+      <% else %>
+        <li class="c-card c-card--interactive flex items-center px-2">
+          <span class="py-1">
+            <% if role.resource_name %>
+              <%= role.resource_name.to_s %>
+            <% else %>
+              <%= role_display_name(role) %>
+            <% end %>
+          </span>
+          <div>
+            <%= link_to url_for(action: :destroy, user_id: @user.id, role: role.name.to_sym, resource_type: role.resource_type), method: :delete, data: { confirm: "Are you sure?" }, class: "c-link c-link--block c-link--icon-alone" do %>
```

Comment:
I need to change this `link_to` to be a `button` so it would likely be some sort of a form like [here](https://github.com/forem/forem/pull/16174/files#diff-cbc52aa9b8fa2a8118a6c022d89188557bd4740d588097fc747dd2da01b75adeR31-R33)?